src/server/hset_family.cc

```diff
@@ -1276,4 +1339,41 @@ int32_t HSetFamily::FieldExpireTime(const DbContext& db_context, const PrimeValu
   }
 }
 
+vector<long> HSetFamily::SetFieldsExpireTime(const OpArgs& op_args, PrimeValue& pv,
+                                             uint32_t ttl_sec, string_view key, CmdArgList values) {
+  DCHECK_EQ(OBJ_HASH, pv.ObjType());
+  vector<long> res;
+  res.reserve(values.size());
+
+  op_args.shard->search_indices()->RemoveDoc(key, op_args.db_cntx, pv);
```

Comment:
I was wrong. We need this because `InitRobj` might change the `inner` pointer which is then used within the `search_indices`. We could conditionally remove only when we create a new `sm` but that's a nit and it won't affect anything :)